cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/TableId.java

```diff
@@ -51,26 +50,6 @@ public static TableId from(Id.Namespace namespaceId, String tableName) {
     return new TableId(namespaceId, tableName);
   }
 
-  /**
-   * Construct a {@link TableId} from a dataset name
-   * TODO: CDAP-1593 - This is bad and should be removed, since it makes assumptions about the dataset name format.
-   * Will need a bigger change to have DatasetAdmin classes accept a namespace id in some other form to pass to 
-   * #from(tablePrefix, namespace, tableName) as opposed to getting it from the {@link DatasetSpecification#getName}
-   *
-   * @param name the dataset/table name to construct the {@link TableId} from
-   * @return the {@link TableId} object for the specified dataset/table name
-   */
-  public static TableId from(String name) {
-    Preconditions.checkArgument(name != null, "Dataset name should not be null");
-    // Dataset/Table name is expected to be in the format <table-prefix>.<namespace>.<name>
-    String invalidFormatError = String.format("Invalid format for dataset/table name '%s'. " +
-                                                "Expected - <table-prefix>.<namespace>.<dataset/table-name>", name);
-    String [] parts = name.split("\\.", 3);
-    Preconditions.checkArgument(parts.length == 3, invalidFormatError);
```

Comment:
Yea, it will be caught by the upgrade tool and logged.
Alternative is: if one part of upgrade tool fails, continue to other parts, but I think the current design is to stop if something is wrong.